squid-blockexplorer/src/blocks/utils.ts

```diff
@@ -125,10 +125,10 @@ export function calcSpacePledged(solutionRange: bigint): bigint {
  * @return {bigint} - size of the history in bytes
  */
 export function calcHistorySize(segmentsCount: number): bigint {
-  const segmentCountBigInt = BigInt(segmentsCount);
-  const PIECES_IN_SEGMENT = BigInt(256);
+  const PIECES_IN_SEGMENT = 256;
+  const PIECE_SIZE = 1048576;
 
-  return PIECE_SIZE * PIECES_IN_SEGMENT * segmentCountBigInt;
+  return BigInt(PIECE_SIZE * PIECES_IN_SEGMENT * segmentsCount);
```

Comment:
Same here

Reply:
Here I'm not using the `Xn` shortcut. Should replace the constructor with it?

Reply:
While it is fairly unlikely that history size will exceed 2^53 any time soon, I'd probably use BigInt anyway. Or else why do you need to return `bigint` in the first place?

Reply:
Why waiting if it literally cost you adding two `n` at the end of constants and wrap `segmentsCount` rather than the whole expression to support every imaginable history size?